### DISCUSSION TITLE
Trivy not scanning Sidecars in K8s cluster

### DISCUSSION THREAD

**Discussion by ChrisJBurns** [aquasecurity/trivy#4173] [Q&A]
## Description

It doesn't seem that Trivy is reporting the images that the sidecars are using for our pods.

## What did you expect to happen?
I expected the Istio proxy image to appear as it's a sidecar

## What happened instead?
We ran:
```shell
 trivy k8s --report summary cluster --kubeconfig ~/.kube/config/our_config --context development_cluster --timeout 30m0s --scanners vuln
```
The image didn't appear on our summary report alongside other sidecars that are being used by our grafana & prometheus deployment.

## Output of run with `-debug`:

```
(paste your output here)
```

## Output of `trivy -v`:

```log
Version: 0.40.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-04-18 12:07:11.190735073 +0000 UTC
  NextUpdate: 2023-04-18 18:07:11.190734673 +0000 UTC
  DownloadedAt: 2023-04-18 12:17:30.092287 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-04-18 00:49:23.663862828 +0000 UTC
  NextUpdate: 2023-04-21 00:49:23.663862428 +0000 UTC
  DownloadedAt: 2023-04-18 12:20:31.341789 +0000 UTC
Policy Bundle:
  Digest: sha256:d19c4c0d48ed4641862e020ff7eba7fd3ba449f66b532b09d79a6023bc65bd5b
  DownloadedAt: 2023-04-18 12:17:31.497949 +0000 UTC
```

**Comment by chen-keinan** [comment]
@ChrisJBurns thank you for reporting this issue , I'll review it and update back

**Comment by chen-keinan** [comment]
@ChrisJBurns I have check this issue and confirm that trivy is scanning also sidecar , the command you are using above is a summary level command,  meaning it aggregate results in `pod/deployment` level , not container.

it you will use the detail (`all`) command for example  , you'll see the details results include sidecar:
```shell
trivy k8s cluster  --report all --format json
```

**Reply by ChrisJBurns** [reply]
Thank you, I can see the information now, although it's a bit much for my use case. For my use case I just want to know what images are running inside of the cluster, this would include Pods, Jobs, Sidecars etc. But the `--report all` seems to generate large amounts of data about binaries also. Is there a reason that summary only includes pod level and not container? As these images get missed otherwise
